cmake/CeleritasUtils.cmake

```diff
@@ -266,45 +304,56 @@ endfunction()
 #-----------------------------------------------------------------------------#
 
 function(celeritas_add_library target)
-  celeritas_rdc_add_library(${target} ${ARGN})
+  
+  cuda_get_sources_and_options(_sources _cmake_options _options ${ARGN})
+  cuda_rdc_sources_contains_cuda(_cuda_sources ${_sources})
+  if(CELERITAS_USE_HIP AND _cuda_sources)
+    # When building Celeritas libraries, we put HIP/CUDA files in shared .cu
+    # suffixed files. Override the language if using HIP.
+    set_source_files_properties(
+      ${_cuda_sources}
+      PROPERTIES LANGUAGE HIP
+    )
+  endif()
+# Potential speed-up (if it is a significant gain should it be applied to the
+# other calls to?)
+#  if(NOT CELERITAS_USE_VecGeom OR NOT CMAKE_CUDA_COMPILER OR NOT _cuda_sources)
+#    add_library(${target} ${ARGN})
+# else()
+  cuda_rdc_add_library(${target} ${ARGN})
+# endif()
 
   # Add Celeritas:: namespace alias
-  add_library(Celeritas::${target} ALIAS ${target})
-
-  set(_targets ${target})
-  get_target_property(_tgt ${target} CELERITAS_CUDA_FINAL_LIBRARY)
-  if(_tgt)
-    celeritas_strip_alias(_tgt ${_tgt})
-    # Building with CUDA RDC support: add final library
-    list(APPEND _targets ${_tgt})
-    get_target_property(_tgt ${target} CELERITAS_CUDA_STATIC_LIBRARY)
-    celeritas_strip_alias(_tgt ${_tgt})
-    if(NOT _tgt STREQUAL target)
-      # Shared and static library have different names
-      list(APPEND _targets ${_tgt})
-    endif()
-    get_target_property(_tgt ${target} CELERITAS_CUDA_OBJECT_LIBRARY)
-    if(_tgt)
-      celeritas_strip_alias(_tgt ${_tgt})
-      set_target_properties(${_tgt} PROPERTIES POSITION_INDEPENDENT_CODE ON)
-    endif()
-  endif()
+  cuda_rdc_add_library(Celeritas::${target} ALIAS ${target})
 
   # Build all targets in lib/
-  set_target_properties(${_targets} PROPERTIES
+  cuda_rdc_set_target_properties(${target} PROPERTIES
+    POSITION_INDEPENDENT_CODE ON   # Technically we probably only need it on the OBJECT library
```

Comment:
Why was this added? PIC should only be added when it's asked for (or if, as you say, we're building object code to be included in a shared library). Nothing in Celeritas itself requires it; it's only needed for ROOT, etc.

Reply:
It was already there.

Reply:
It was introduced by 7ea96499838f8a006fa91f6374fcc79d4e60d4e3 but thanks to the squashing it is unclear why it is there.

Reply:
I know why it's required for object library, but it looks like this refector removed it from an "if (object library)" condition.

Reply:
PS you can look at the unsquash commits: https://github.com/celeritas-project/celeritas/pull/545/commits
PPS some aspects of this PR would've been easier to review with a separate granular commit for "find replace CELERITAS_RDC"

Reply:
Yes, that was a thinko.  The thinking was that if there is an object library then it is used to build the rest and so it does not hurt to have it applied on all ... but the `if` got left out :(

Reply:
> PPS some aspects of this PR would've been easier to review with a separate granular commit for "find replace CELERITAS_RDC"

Sorry.  My side-view of things (through the source repository) actually has this commit in per se.  I can easily rewrite the history if you prefer.

Reply:
Source repository? Nah, don't bother about it. If you really want you could
```console
$ git reset $(git commit-tree -m "Merge other history" -p HEAD -p other_sha HEAD^{tree})
```

Reply:
Source repository: https://github.com/pcanal/CudaRDCExercise :)

Reply:
Oh gotcha. Definitely don't worry about the history then.